includes/payment-methods/class-upe-payment-gateway.php

```diff
@@ -86,36 +86,41 @@ public function __construct(
-		$this->title              = __( 'WooCommerce Payments', 'woocommerce-payments' );
 		$this->description        = '';
-		$this->checkout_title     = __( 'Popular payment methods', 'woocommerce-payments' );
-		$this->payment_methods    = $payment_methods;
+		$this->stripe_id          = $payment_method->get_id();
+		$this->payment_method     = $payment_method;
+		$this->title              = $payment_method->get_title();
 
-		add_action( 'wc_ajax_wcpay_create_payment_intent', [ $this, 'create_payment_intent_ajax' ] );
-		add_action( 'wc_ajax_wcpay_update_payment_intent', [ $this, 'update_payment_intent_ajax' ] );
-		add_action( 'wc_ajax_wcpay_init_setup_intent', [ $this, 'init_setup_intent_ajax' ] );
-		add_action( 'wc_ajax_wcpay_log_payment_error', [ $this, 'log_payment_error_ajax' ] );
+		add_action( "wc_ajax_wcpay_create_payment_intent_$this->stripe_id", [ $this, 'create_payment_intent_ajax' ] );
+		add_action( "wc_ajax_wcpay_update_payment_inten_ $this->stripe_id", [ $this, 'update_payment_intent_ajax' ] );
+		add_action( "wc_ajax_wcpay_init_setup_intent_$this->stripe_id", [ $this, 'init_setup_intent_ajax' ] );
 
-		add_action( 'wp_ajax_save_upe_appearance', [ $this, 'save_upe_appearance_ajax' ] );
-		add_action( 'wp_ajax_nopriv_save_upe_appearance', [ $this, 'save_upe_appearance_ajax' ] );
-		add_action( 'switch_theme', [ $this, 'clear_upe_appearance_transient' ] );
-		add_action( 'woocommerce_woocommerce_payments_updated', [ $this, 'clear_upe_appearance_transient' ] );
+		if ( 'card' !== $this->stripe_id ) {
+			$this->id           = self::GATEWAY_ID . '_' . $this->stripe_id;
+			$this->method_title = "WooCommerce Payments ($this->title)";
+		} else {
+			// Only add these filters once.
+			add_action( 'wc_ajax_wcpay_log_payment_error', [ $this, 'log_payment_error_ajax' ] );
+			add_action( 'wp_ajax_save_upe_appearance', [ $this, 'save_upe_appearance_ajax' ] );
+			add_action( 'wp_ajax_nopriv_save_upe_appearance', [ $this, 'save_upe_appearance_ajax' ] );
+			add_action( 'switch_theme', [ $this, 'clear_upe_appearance_transient' ] );
+			add_action( 'woocommerce_woocommerce_payments_updated', [ $this, 'clear_upe_appearance_transient' ] );
 
-		add_action( 'wp', [ $this, 'maybe_process_upe_redirect' ] );
+			add_action( 'wp', [ $this, 'maybe_process_upe_redirect' ] );
 
-		if ( ! is_admin() ) {
-			add_filter( 'woocommerce_gateway_title', [ $this, 'maybe_filter_gateway_title' ], 10, 2 );
-		}
+			add_action( 'woocommerce_order_payment_status_changed', [ __CLASS__, 'remove_upe_payment_intent_from_session' ], 10, 0 );
+			add_action( 'woocommerce_after_account_payment_methods', [ $this, 'remove_upe_setup_intent_from_session' ], 10, 0 );
 
-		add_action( 'woocommerce_order_payment_status_changed', [ __CLASS__, 'remove_upe_payment_intent_from_session' ], 10, 0 );
-		add_action( 'woocommerce_after_account_payment_methods', [ $this, 'remove_upe_setup_intent_from_session' ], 10, 0 );
+			if ( is_admin() ) {
```

Comment:
@FangedParakeet @mdmoore Is this supposed to be `! is_admin()`?

Reply:
I feel like this had something to do with preventing the UPE gateway titles from being changed but ill defer to @FangedParakeet.

Reply:
I think this _is_ actually supposed to be `is_admin()`. What happened here is that I repurposed the function `maybe_filter_gateway_title`, but neglected to update the comment describing it.

Previously the function `maybe_filter_gateway_title` was intended to be used at checkout. Since there was only a single payment gateway, containing multiple payment methods (in the UPE), we used `maybe_filter_gateway_title` to title the single gateway dynamically. If there were no payment methods enabled, we'd default to the card title, if only a single payment method were enabled we'd title the gateway the name of that payment method, and if multiple payment methods were enabled the gateway would be titled "Popular payment methods".

However, in this brave new realisation, each payment method has its own gateway at checkout, so we don't need to rename anything at checkout. Instead I am renaming the gateway in the WP Admin settings to be titled "WooCommerce Payments (`{{ Payment method title }}`)". That's why `maybe_filter_gateway_title` basically returns the property `method_title` instead of the actual `title`. 

Having explained that, I actually think this entire function is now redundant. When I was working on this, I think I originally had all the payment methods appearing as separate gateways in WP Admin (i.e. if you go to **Payments > Settings > All payment methods**). This is how the payment methods appear in the Stripe gateway, but I think after chatting to @LevinMedia (RIP), I changed this (or I did nothing and it happened by accident) so that only WooCommerce Payments would appear on this screen and users could only manage the enabled gateways through the WCPay Settings. 

**TL;DR**
Basically this wasn't a typo, but I think we can delete this whole thing--function included--anyways and that'll also be one less unit test you'll have to worry about fixing (lol).

Reply:
Sweeeet! Thanks for the explanation! I will take a look and remove it